### ISSUE TITLE
cvc5 suffered a segfault when producing proofs in experimental format

### ISSUE THREAD

**Issue by rodrigo7491** [cvc5/cvc5#10836]
**Describe the bug**
When trying to produce alethe proofs with the attached benchmark (see [test.zip](https://github.com/user-attachments/files/15514369/test.zip)), the output is the following:
```
unsat
cvc5 suffered a segfault.
Offending address is 0x4
Looks like a NULL pointer was dereferenced.
Segmentation fault (core dumped)
```
This does not occur with lfsc proofs nor with aletheLF proofs.

**Command line arguments**: `--proof-format=alethe --proof-granularity=theory-rewrite`
**cvc5 version/commit**: v1.1.2
**Operating system**: Linux Mint 21.2

**Comment by HanielB** [comment]
Hi Rodrigo. Alethe proof support in main is currently unstable and is expected to have issues, which is always it requires the option `--proof-alethe-experimental` for the moment (if you use `--dump-proofs`, for example, it'll give an error message unless you pass that option). In current man (`1.1.3-dev.297.c5ba0da878`) I cannot reproduce your issue though.

FYI in [this branch](https://github.com/cvc5/cvc5/tree/proof-new) the cvc5 support for Alethe is robust and in particular it produces a valid proof for the problem you posted.

**Comment by rodrigo7491** [comment]
Hi Haniel. Thanks for explaining the situation and letting me know about the `proof-new` branch. I've tried different things and here is the situation on my side.

1. With the latest release, v1.1.2, I get the behaviour I mentioned above.
2. With the current main, `c5ba0da`, I get a different message but the same error:
```
unsat
cvc5 suffered a segfault.
Offending address is 0x557b81b7003a
Segmentation fault (core dumped)
```
3. With the current version on the `proof-new` branch a proof is produced successfully.

The `--proof-alethe-experimental` flag seems to be new, since it is not listed in the `--help` of v1.1.2 and when used there it gives the following message:
```
(error "Error in option parsing: can't understand option `--proof-alethe-experimental'")

Please use --help to get help on command-line options.
```
In any case, it seems the issue I reported will be fixed once `proof-new` is merged.

**Comment by rodrigo7491** [comment]
Just for the record (and maybe not worth opening a new issue), I encountered another problem with v1.1.2. With [another_test.zip](https://github.com/user-attachments/files/15584604/another_test.zip) I get the following output when trying to produce alethe proofs:
```
unsat
Fatal failure within virtual cvc5::internal::Node cvc5::internal::proof::AletheNodeConverter::postConvert(cvc5::internal::Node) at /home/rodrigo/Downloads/test_cvc_segfault/cvc5-main/src/proof/alethe/alethe_node_converter.cpp:36
Unreachable code reached Fresh Skolems are not allowed

Aborted (core dumped)
```
I observed this behaviour with v1.1.2 as well as with main (`c5ba0da`), but not with the `proof-new` branch.

**Comment by HanielB** [comment]
Thanks, @rodrigo7491. This is the same issue as #9927. In the coming weeks as the fixes from `proof-new` are merged this will be addressed.

**Comment by HanielB** [comment]
This issue has been fixed in main with the recent update of the Alethe backend.
